source/server/configuration_impl.cc

```diff
@@ -14,13 +14,14 @@
 namespace Server {
 namespace Configuration {
 
-void FilterChainUtility::buildFilterChain(Network::FilterManager& filter_manager,
+bool FilterChainUtility::buildFilterChain(Network::FilterManager& filter_manager,
                                           const std::list<NetworkFilterFactoryCb>& factories) {
   for (const NetworkFilterFactoryCb& factory : factories) {
     factory(filter_manager);
   }
 
   filter_manager.initializeReadFilters();
+  return factories.size() > 0;
```

Comment:
There is a more reliable way to check that the filter chain is populated
- Check  `upstream_filters_` isn't empty (from here: https://github.com/lyft/envoy/blob/7e57daa9c6cb82c936c34bf5470b1a7508798f6d/source/common/network/filter_manager_impl.h#L67). This can be done by exposing a method on FilterManagerImpl
or
- Have  NetworkFilterFactoryCb::factory return a bool and check that as you loop through